packages/flutter_tools/lib/src/base/analyze_size.dart

```diff
@@ -0,0 +1,394 @@
+    }
+
+    final SymbolNode apkAnalysisRoot = _parseUnzipFile(unzipOut);
+
+    for (final SymbolNode firstLevelPath in apkAnalysisRoot.children) {
+      _printEntitySize(
+        firstLevelPath.name,
+        firstLevelPath.value,
+        1,
+      );
+      if (firstLevelPath.name == 'lib') {
+        _printLibDetails(firstLevelPath, '', aotSizeJson);
+      }
+    }
+
+    _logger.printStatus('▒' * tableWidth);
+    
+    Map<String, dynamic> apkAnalysisJson = apkAnalysisRoot.toJson();
+    
+    // TODO(peterdjlee): Add aot size for all platforms.
+    apkAnalysisJson = addAotSizeDataToJson(
+      apkAnalysisJson,
+      'lib/arm64-v8a/libapp.so (Dart AOT)'.split('/'), 
+      json.decode(aotSizeJson.readAsStringSync()) as List<dynamic>,
+    );
+
+    // Rearrange children to ensure the type key shows at the top.
+    final List<dynamic> rootChildren = apkAnalysisJson['children'] as List<dynamic>;
+    apkAnalysisJson.remove('children');
+    
+    apkAnalysisJson['type'] = 'apk';
+    apkAnalysisJson['children'] = rootChildren;
+
+    return apkAnalysisJson;
+  }
+
+  // Parse the output of unzip -v which shows the zip's contents' compressed sizes.
+  // Example output of unzip -v:
+  //  Length   Method    Size  Cmpr    Date    Time   CRC-32   Name
+  // --------  ------  ------- ---- ---------- ----- --------  ----
+  //    11708  Defl:N     2592  78% 00-00-1980 00:00 07733eef  AndroidManifest.xml
+  //     1399  Defl:N     1092  22% 00-00-1980 00:00 f53d952a  META-INF/CERT.RSA
+  //    46298  Defl:N    14530  69% 00-00-1980 00:00 17df02b8  META-INF/CERT.SF
+
+  final RegExp _parseUnzipOutput = RegExp(r'^\s*\d+\s+[\w|:]+\s+(\d+)\s+.*  (.+)$');
+
+  SymbolNode _parseUnzipFile(String unzipOut) {
+    final Map<List<String>, int> pathsToSize = <List<String>, int>{};
+
+    // Parse each path into pathsToSize so that the key is a list of
+    // path parts and the value is the size.
+    // For example:
+    // 'path/to/file' where file = 1500 => pathsToSize[['path', 'to', 'file']] = 1500
+    for (final String line in const LineSplitter().convert(unzipOut)) {
+      final RegExpMatch match = _parseUnzipOutput.firstMatch(line);
+      if (match == null) {
+        continue;
+      }
+      pathsToSize[match.group(2).split('/')] = int.parse(match.group(1));
+    }
+
+    final SymbolNode rootNode = SymbolNode('Root');
+
+    SymbolNode currentNode = rootNode;
+    for (final List<String> paths in pathsToSize.keys) {
+      for (final String path in paths) {
+        SymbolNode childWithPathAsName = currentNode.childByName(path);
+
+        if (childWithPathAsName == null) {
+          childWithPathAsName = SymbolNode(path);
+
+          if (path.endsWith('libapp.so')) {
+            childWithPathAsName.name += ' (Dart AOT)';
+          } else if (path.endsWith('libflutter.so')) {
+            childWithPathAsName.name += ' (Flutter Engine)';
+          }
+          currentNode.addChild(childWithPathAsName);
```

Comment:
does `childWithPathAsName.addValue(pathsToSize[paths]);` need to be called before `currentNode.addChild(childWithPathAsName);` so that the `childWithPathAsName's` value is summed into `currentNode's` value when calling `addChild`?